assets/src/edit-story/components/form/advancedDropDown/list/index.js

```diff
@@ -223,36 +221,49 @@ function FontList({
               role="group"
               aria-labelledby={groupLabelId}
             >
-              <GroupLabel id={groupLabelId} role="presentation">
-                {group.label}
-              </GroupLabel>
-              {group.options.map((font, j) => (
-                <Option
-                  key={font.name}
-                  ref={(el) =>
-                    (optionsRef.current[
-                      getInset(filteredListGroups, i, j)
-                    ] = el)
-                  }
-                  role="option"
-                  tabIndex={-1}
-                  aria-selected={value === font.name}
-                  aria-posinset={getInset(filteredListGroups, i, j)}
-                  aria-setsize={filteredOptions.length}
-                  data-font={font.name}
-                  onClick={() => onSelect(font.name)}
-                  fontFamily={
-                    font.service.includes('google')
-                      ? `'${font.name}::MENU'`
-                      : font.name
-                  }
-                >
-                  {value === font.name && (
-                    <Selected aria-label={__('Selected', 'web-stories')} />
-                  )}
-                  {font.name}
-                </Option>
-              ))}
+              {group.label && (
+                <GroupLabel id={groupLabelId} role="presentation">
+                  {group.label}
+                </GroupLabel>
+              )}
+              {group.options.map((option, j) => {
+                const optionProps = {
+                  key: option.id,
+                  role: 'option',
+                  tabIndex: -1,
+                  'aria-selected': value === option.id,
+                  'aria-posinset': getInset(filteredListGroups, i, j),
+                  'aria-setsize': filteredOptions.length,
+                  'data-option': option.id,
+                  onClick: () => onSelect(option.id),
+                };
+                return renderer ? (
+                  <OptionRenderer
+                    {...optionProps}
+                    option={option}
+                    optionRef={(el) =>
+                      (optionsRef.current[
+                        getInset(filteredListGroups, i, j)
+                      ] = el)
+                    }
+                    onClick={() => onSelect(option.id)}
+                  />
+                ) : (
+                  <Option
+                    {...optionProps}
+                    ref={(el) =>
+                      (optionsRef.current[
+                        getInset(filteredListGroups, i, j)
+                      ] = el)
+                    }
+                  >
+                    {value === option.id && (
+                      <Selected aria-label={__('Selected', 'web-stories')} />
+                    )}
+                    {option.name}
+                  </Option>
+                );
```

Comment:
Please move this to a separate component to clean this up a bit. And can't the `<OptionRenderer />` get a normal `ref` too, if you make it use `forwardRef()`?